src/Symfony/Bundle/FrameworkBundle/Client.php

```diff
@@ -104,6 +107,46 @@ public function enableReboot()
         $this->reboot = true;
     }
 
+    /**
+     * Logins a user against a specific firewall.
+     *
+     * @param string|object $user
+     * @param string[]      $roles
+     * @param string        $firewallName
+     * @param string|null   $firewallContext
+     */
+    public function login(
+        $user,
+        array $roles = array(),
+        string $firewallName = 'main',
+        string $firewallContext = null
```

Comment:
If you pass the firewall-name shouldn't this read the configuration of the application? Rather then having to pass the context, which could be different in the actual application config, making your test pass but fail in a real-life situation.

Reply:
I didn't think about that, good catch :+1: